rdmd.d

```diff
@@ -916,3 +898,78 @@ void yap(size_t line = __LINE__, T...)(auto ref T stuff)
     debug stderr.writeln(line, ": ", stuff);
     else stderr.writeln(stuff);
 }
+
+struct FileAttributes
+{
+    uint attributes;
+    bool exists() const { return attributes != 0xFFFFFFFF; }
+    bool isFile() const
+        in { assert(exists); } do
+    {
+        import std.file : attrIsFile;
+        return attrIsFile(attributes);
+    }
+}
+
+struct Yap
+{
+    static FileAttributes getFileAttributes(const(char)[] name)
+    {
+        yap("stat ", name);
+        import std.internal.cstring : tempCString;
+        version(Windows)
+        {
+            import core.sys.windows.windows : GetFileAttributesW;
+            auto attributes = GetFileAttributesW(name.tempCString!wchar());
+            return FileAttributes(attributes);
+        }
+        else version(Posix)
+        {
+            import core.sys.posix.sys.stat : stat_t, stat;
+            stat_t statbuf = void;
+            return FileAttributes(
+                (0 == stat(name.tempCString(), &statbuf)) ? statbuf.st_mode : 0xFFFFFFFF);
+        }
+        else static assert(0);
+    }
+    static bool exists(const(char)[] name)
+    {
+        return Yap.getFileAttributes(name).exists();
+    }
+    static bool existsAsFile(const(char)[] name)
+    {
+        auto attributes = Yap.getFileAttributes(name);
+        return attributes.exists() && attributes.isFile();
+    }
+    static auto timeLastModified(R)(R name)
+    {
+        yap("stat ", name);
+        return std.file.timeLastModified(name);
```

Comment:
This could also be written as simply `return .timeLastModified(name);`, but I guess it's probably better to be more explicit here.